src/components/views/elements/SettingsFlag.tsx

```diff
@@ -104,14 +95,13 @@ export default class SettingsFlag extends React.Component<IProps, IState> {
     };
 
     public render(): React.ReactNode {
-        const canChange = SettingsStore.canSetValue(this.props.name, this.props.roomId ?? null, this.props.level);
+        const disabled = !SettingsStore.canSetValue(this.props.name, this.props.roomId ?? null, this.props.level);
 
-        if (!canChange && this.props.hideIfCannotSet) return null;
+        if (disabled && this.props.hideIfCannotSet) return null;
 
         const label = this.props.label ?? SettingsStore.getDisplayName(this.props.name, this.props.level);
         const description = SettingsStore.getDescription(this.props.name);
         const shouldWarn = SettingsStore.shouldHaveWarning(this.props.name);
-        const disabled = this.state.disabled || !canChange;
```

Comment:
note that it was impossible for `canChange` to be true while `state.disabled` was true, so this was equivalent to `const disabled = !canChange;`.